src/Arcus.Security.Tests.Integration/HashiCorp/Hosting/HashiCorpVaultTestServer.cs

```diff
@@ -292,9 +336,18 @@ private void StopHashiCorpVault()
 
         protected PolicyResult RetryAction(Action action, int timeoutInSeconds = 30, int retryIntervalInSeconds = 1)
         {
-            Guard.NotNull(action, nameof(action), "Requires disposing function to be retried");
-            Guard.NotLessThan(timeoutInSeconds, 0, nameof(timeoutInSeconds), "Requires a timeout (in sec) greater than zero");
-            Guard.NotLessThan(retryIntervalInSeconds, 0, nameof(retryIntervalInSeconds), "Requires a retry interval (in sec) greater than zero");
+            if (action is null)
+            {
+                throw new ArgumentNullException(nameof(action), "Requires disposing function to be retried");
+            }
+            if (timeoutInSeconds < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(timeoutInSeconds), "Requires a timeout (in sec) greater than zero");
+            }
+            if (retryIntervalInSeconds < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(retryIntervalInSeconds), "Requires a retry interval (in sec) greater than zero");
+            }
```

Comment:
```suggestion
            if (action is null)
            {
                throw new ArgumentNullException(nameof(action), "Requires disposing function to be retried");
            }

            if (timeoutInSeconds < 0)
            {
                throw new ArgumentOutOfRangeException(nameof(timeoutInSeconds), "Requires a timeout (in sec) greater than zero");
            }

            if (retryIntervalInSeconds < 0)
            {
                throw new ArgumentOutOfRangeException(nameof(retryIntervalInSeconds), "Requires a retry interval (in sec) greater than zero");
            }
```
Best to have a blank line between curly brackets, following the Microsoft code style.